.github/workflows/push_docker_image.yml

```diff
@@ -0,0 +1,21 @@
+name: push docker image
+
+on: 
+  push:
+    branches:
+      - main
+
+jobs:
+  build_and_push:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: login
+        run: docker login --username ${{ secrets.DOCKER_USERNAME }} --password ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
```

Comment:
Have we added these secrets to this upstream repository?

Reply:
We should add them *before* merging this.

Reply:
```suggestion
        run: docker login --username '${{ secrets.DOCKER_USERNAME }}' --password '${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}'
```

Reply:
ok i'll add the secrets right now.